File: tests/integration/construct_component_unknown/python/resource.py
# Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

from typing import Any, Optional

import khulnasoft
import khulnasoft.dynamic as dynamic


_ID = 0


class MyDynamicProvider(dynamic.ResourceProvider):
    def create(self, props: Any) -> dynamic.CreateResult:
        global _ID
        _ID = _ID + 1
        return dynamic.CreateResult(id_=str(_ID))


class Resource(dynamic.Resource):
    def __init__(self, name: str, opts: Optional[khulnasoft.ResourceOptions]=None):
        super().__init__(MyDynamicProvider(), name, {}, opts)

File: tests/integration/partial_values/nodejs/resource.ts
// Copyright 2016-2018, Pulumi Corporation.  All rights reserved.

import * as khulnasoft from "@khulnasoft/khulnasoft";

let currentID = 0;

export class Provider implements khulnasoft.dynamic.ResourceProvider {
    public static readonly instance = new Provider();

    public readonly create: (inputs: any) => Promise<khulnasoft.dynamic.CreateResult>;

    constructor() {
        this.create = async (inputs: any) => {
            return {
                id: (currentID++).toString(),
                outs: inputs,
            };
        };
    }
}

export class Resource extends khulnasoft.dynamic.Resource {
    public readonly foo: khulnasoft.Output<string>;
    public readonly bar: khulnasoft.Output<{ value: string, unknown: string }>;
    public readonly baz: khulnasoft.Output<any[]>;

    constructor(name: string, props: ResourceProps, opts?: khulnasoft.ResourceOptions) {
        super(Provider.instance, name, props, opts);
    }
}

export interface ResourceProps {
    foo: khulnasoft.Input<string>;
    bar: khulnasoft.Input<{ value: khulnasoft.Input<string>, unknown: khulnasoft.Input<string> }>;
    baz: khulnasoft.Input<khulnasoft.Input<any>[]>;
}
